src/common/metrics_context.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <string>
+
+namespace shar {
+
+struct MetricsContext {
```

Comment:
It is not really a context, more like `MetricDescription` (also, singular form).